### ISSUE TITLE
Terminate user state

### ISSUE THREAD

**Issue by vit1251** [python-poetry/poetry#5827]
During setup dependencies I meet with `TooManyRedirects` exceptions and it terminate state (see below more deep).
I start poetry with `-vvv` options to receive more information but does it does not resolve my issue.
Could you please show more details about that redirects and suggest way (from poetry output) how to solve dependencies problme.

```
  TooManyRedirects

  Exceeded 30 redirects.

  at /usr/lib/python3/dist-packages/requests/sessions.py:165 in resolve_redirects
      161│             except (ChunkedEncodingError, ContentDecodingError, RuntimeError):
      162│                 resp.raw.read(decode_content=False)
      163│ 
      164│             if len(resp.history) >= self.max_redirects:
    → 165│                 raise TooManyRedirects('Exceeded %s redirects.' % self.max_redirects, response=resp)
      166│ 
      167│             # Release the connection back into the pool.
      168│             resp.close()
```

**Comment by Secrus** [comment]
This is upstream issue, as PyPI API was broken. This is getting fixed at the moment. Check https://status.python.org/ for status updates.

**Comment by vit1251** [comment]
OK. Thanks.

**Comment by christophe-david** [comment]
Thank you for posting this before I do, though I'm concerned about the problem itself.
It looks like a global problem to me. I get this during a `poetry install` with a `poetry.lock` that worked perfectly fine yesterday.
Today, it fails on my computer and in my GitHub workflows.
I tried clearing Poetry cache, but to no avail.
Also, `requests` got updated yesterday, so I wonder if it can be related somehow...
